examples/metrics_simple/README.md

```diff
@@ -1,86 +1,108 @@
+    ```
 
-2: Initialize a MeterProvider and add the reader.
+2. Initialize a MeterProvider and add the reader.
 We will use this to obtain Meter objects in the future.
 
-```cpp
-auto provider = std::shared_ptr<metrics_api::MeterProvider>(new opentelemetry::metrics::MeterProvider());
-auto p = std::static_pointer_cast<metric_sdk::MeterProvider>(provider);
-p->AddMetricReader(std::move(reader));
-```
-
-3: Create and add a view to the provider.
-
-```cpp
-std::unique_ptr<metric_sdk::InstrumentSelector> instrument_selector{
-    new metric_sdk::InstrumentSelector(metric_sdk::InstrumentType::kCounter, "name_counter")};
-std::unique_ptr<metric_sdk::MeterSelector> meter_selector{
-    new metric_sdk::MeterSelector(name, version, schema)};
-std::unique_ptr<metric_sdk::View> sum_view{
-    new metric_sdk::View{name, "description", metric_sdk::AggregationType::kSum}};
-p->AddView(std::move(instrument_selector), std::move(meter_selector), std::move(sum_view));
-```
-
-4: Then create a
-[Counter](https://github.com/open-telemetry/opentelemetry-specification/blob/main/specification/metrics/api.md#counter)
-instrument from it. Every Meter pointer returned by the
-MeterProvider points to the same Meter. This means that the Meter will be able
-to combine metrics captured from different functions without having to
-constantly pass the Meter around the library.
-
-```cpp
-nostd::shared_ptr<metrics_api::Meter> meter = provider->GetMeter(name, "1.2.0");
-auto double_counter = meter->CreateDoubleCounter(counter_name);
-// Create a label set which annotates metric values
-std::map<std::string, std::string> labels = {{"key", "value"}};
-auto labelkv = common::KeyValueIterableView<decltype(labels)>{labels};
-double_counter->Add(val, labelkv);
-```
-
-5: To use histogram instrument, a view with proper `InstrumentType` and `AggregationType`
-has to be added to the provider.
-
-```cpp
-std::unique_ptr<metric_sdk::InstrumentSelector> histogram_instrument_selector{
-    new metric_sdk::InstrumentSelector(metric_sdk::InstrumentType::kHistogram, "histogram_name")};
-std::unique_ptr<metric_sdk::MeterSelector> histogram_meter_selector{
-    new metric_sdk::MeterSelector(name, version, schema)};
-std::unique_ptr<metric_sdk::View> histogram_view{
-    new metric_sdk::View{name, "description", metric_sdk::AggregationType::kHistogram}};
-p->AddView(std::move(histogram_instrument_selector), std::move(histogram_meter_selector),
-            std::move(histogram_view));
-
-auto histogram_counter = meter->CreateDoubleHistogram("histogram_name");
-auto context = opentelemetry::context::Context{};
-histogram_counter->Record(val, labelkv, context);
-```
-
-See [CONTRIBUTING.md](../../CONTRIBUTING.md) for instructions on building and
+    ```cpp
+    auto provider = std::shared_ptr<metrics_api::MeterProvider>(new opentelemetry::metrics::MeterProvider());
+    auto p = std::static_pointer_cast<metric_sdk::MeterProvider>(provider);
```

Comment:
`new opentelemetry::metrics::MeterProvider()`

Does this builds ?
The API class is abstract.

How about instead:

    auto provider = std::shared_ptr<metrics_sdk::MeterProvider>(new opentelemetry::sdk::metrics::MeterProvider());

which also avoids the downcast from API to SDK.

Reply:
Good catch. Have changed it accordingly.